lib/core.ex

```diff
@@ -124,34 +124,61 @@ defmodule TelemetryMetricsPrometheus.Core do
           | {:validations, Registry.validation_opts() | false}
 
   @doc """
-  Initializes a reporter instance with the provided `Telemetry.Metrics` definitions.
+  Reporter's child spec.
+
+  This function allows you to start the reporter under a supervisor like this:
+
+  children = [
+    {TelemetryMetricsPrometheus.Core, options}
+  ]
+
+  See `init/1` for a list of available options.
 
   Available options:
   * `:name` - name of the reporter instance. Defaults to `:prometheus_metrics`
   * `:monitor_reporter` - collects metrics on the reporter's ETS table usage. Defaults to `false`
   * `:validations` - Keyword options list to control validations. All validations can be disabled by setting `validations: false`.
-    * `:consistent_units` - logs a warning when mixed time units are found in your definitions. Defaults to `true`
-    * `:require_seconds` - logs a warning if units other than seconds are found in your definitions. Defaults to `true`
+  * `:consistent_units` - logs a warning when mixed time units are found in your definitions. Defaults to `true`
+  * `:require_seconds` - logs a warning if units other than seconds are found in your definitions. Defaults to `true`
+  * `:metrics` - a list of metrics to track.
+  """
+  @spec child_spec(prometheus_options()) :: Supervisor.child_spec()
+  def child_spec(options) do
+    opts = ensure_options(options)
+    opts = add_internal_metrics(opts)
+    TelemetryMetricsPrometheus.Core.Registry.child_spec(opts)
```

Comment:
I went back and forth on this a bit but what do you think about manually constructing our spec here and set the start to use the name for the id and `Registry` as the module to start or having a function here that relays the call? It's not a big deal, but I'd rather not leak the Registry out since it's an implementation detail.